frontend/src/citizen-frontend/applications/editor/unit-preference/UnitsSubSection.tsx

```diff
@@ -242,6 +246,25 @@ export default React.memo(function UnitsSubSection({
   )
 })
 
+function updatePreferredUnits(
+  prev: PreferredUnit[],
+  next: PreferredUnit[]
+): PreferredUnit[] {
+  // <MultiSelect> doesn't preserve the order in which items are selected, so this function tries to do it manually.
+  // In practice, the items should be added/removed one at a time, but there's limited support here to handle
+  // additions/removals of multiple items at once.
+
+  if (next.length > prev.length) {
+    const added = next.filter(
+      (u) => prev.find((u2) => u2.id === u.id) === undefined
+    )
+    return [...prev, ...added]
+  } else if (next.length < prev.length) {
+    return prev.filter((u) => next.find((u2) => u2.id === u.id) !== undefined)
+  }
+  return prev
```

Comment:
might be safer to return `next` in case somehow the selection gets swapped instead of separate remove+add